src/js_sync.hpp

```diff
@@ -534,6 +534,8 @@ std::string SessionClass<T>::get_connection_state_value(SyncSession::ConnectionS
         case SyncSession::ConnectionState::Disconnected: return "disconnected";
         case SyncSession::ConnectionState::Connecting: return "connecting";
         case SyncSession::ConnectionState::Connected: return "connected";
+		default:
```

Comment:
You turn a compile time error into a runtime error by making a default here. That doesn't seem useful?

Reply:
There is not compile time error for this. It's a warning _'not all control paths return a value'_. It's best practice imo,  every such switch statement to include a default which either throws or DEBUG asserts. Otherwise if  a new member is added to the enumeration it would introduce strange bugs in the code.

Reply:
Agree - except when it's an enum! Isn't this an enum?
When switching on an enum it's better to not have a default as that will give a compile time warning.
If you get the other warning return something at the end or use our macro  REALM_UNREACHABLE  (I think it's called)

Reply:
My intent with this code change is to make the switch statement fail. I don't want to make the function fail. Slight but important difference. I will change the code to use REALM_UNREACHABLE  in the default statement

Reply:
@blagoev To me the important difference if failing at compile time versus runtime. We should make that work. I really don't like we make the code worse (in my eyes) than it was and would suggest to take this change and discussion in a separate PR to not hold up this one.

Reply:
Given that this code change consist of a check for unknown enumeration member I don't see how its making the code worse. It adds a check which was not there before. We can debate over runtime vs compile time checking and  how to do it better. But this change should stay since it improves over the previous state of the code. I don't see how this is blocking the PR except that you don't like how it's done. Unfortunately this is how its done on many other places in this project code base which consist of the similar switch statements. I would suggest you open a separate issue where we discuss and define what's best and target every single place similar code is present in the code base. I think this discussion is getting outside the PR intent.

Reply:
we agreed with Brian to do a discussion later and define the best strategy on this type of code and fix all places this is used. Resolving for now.

Reply:
https://github.com/realm/realm-js/issues/2751